src/stp/computations/PositionComputations.cpp

```diff
@@ -0,0 +1,324 @@
+                auto intersect = line.intersects(ball2GoalLine);
+                if (intersect.has_value()) lineBorderIntersects.push_back(intersect.value());
+            }
+        }
+
+        // Always use the first (as there should only be one.
+        Vector2 lineBorderIntersect = lineBorderIntersects.front();
+
+        interface::Input::drawData(interface::Visual::DEBUG,std::vector({Vector2(ball2GoalLine.start.x,ball2GoalLine.start.y),Vector2(ball2GoalLine.end.x,ball2GoalLine.end.y)}),Qt::blue,-1,interface::Drawing::LINES_CONNECTED);
+        interface::Input::drawData(interface::Visual::DEBUG,std::vector({lineBorderIntersect}),Qt::red,-1,interface::Drawing::CIRCLES, 3 ,3, 2);
+
+        // DEBUG
+        if (lineBorderIntersects.empty() || lineBorderIntersects.size() > 1) {
+            //RTT_DEBUG("determineWallPositions broke. Size is " + std::to_string(lineBorderIntersects.size()) + "!...");
+        } else {
+            //RTT_DEBUG("Intersect is at " + std::to_string(lineBorderIntersect.x) + "," +std::to_string(lineBorderIntersect.y));
+            int i = 1;
+            if (amountDefenders % 2) {
+                /// ODD
+                positions.push_back(lineBorderIntersect);
+                while (positions.size() < amountDefenders) {
+                    auto circle = Circle(lineBorderIntersect, (i++) * (spacingRobots));
+                    for (const LineSegment &line : defenseAreaBorder) {
+                        auto intersects = intersectsCircleWithLineSegment(circle, line);
+                        for (auto intersect : intersects) {
+                            interface::Input::drawData(interface::Visual::DEBUG, std::vector({intersect}), Qt::yellow,
+                                                       -1, interface::Drawing::CIRCLES, 3, 3, 2);
+                            positions.push_back(intersect);
+                        }
+                    }
+                }
+            } else {
+                /// EVEN
+                while (positions.size() < amountDefenders) {
+                    auto circle = Circle(lineBorderIntersect, (-0.5+i++) * (spacingRobots));
+                    for (const LineSegment &line : defenseAreaBorder) {
+                        auto intersects = intersectsCircleWithLineSegment(circle, line);
+                        for (auto intersect : intersects) {
+                            interface::Input::drawData(interface::Visual::DEBUG, std::vector({intersect}), Qt::yellow,
+                                                       -1, interface::Drawing::CIRCLES, 3, 3, 2);
+                            positions.push_back(intersect);
+                        }
+                    }
+                }
+                }
+            }
+        }
+        return positions;
+    }
+
+    double sq(double x) {
+        return x * x;
+    }
+
+    std::vector<Vector2> PositionComputations::intersectsCircleWithLineSegment(Circle circle, LineSegment line){
```

Comment:
1. This function should be in roboteam_utils.
2. This function is probably copied from the internet 
3. this implementation is pretty inefficient 
4. It's difficult to read

Reply:
This was ment as a temp solution to the problem. Thought we discussed this haha

Reply:
I think I specifically said that it's fine for testing, I didn't say you could PR it ^^

Reply:
Moved it to utils and changed all possible auto's to double's. Do we actually want to change the implementation now? I dont feel like its of high enough importance now to delay the merging of the PR, and personally dont know if i could nor do i feel like improving it

Reply:
Ehh, I don't like having copied code; I don't know *where* you copied it from, but if it's licensed then we would have an issue. So, it depends. You have to remember this is public code.

Reply:
We used this: https://rosettacode.org/wiki/Line_circle_intersection. They have a GNU license, which we should include next to that part of the code (or over our whole project, not sure about that). Do you know if the SSL doesn't allow this license? If not, then that should be fine